### ISSUE TITLE
Investigate more reliable method of detecting controllers

### ISSUE THREAD

**Issue by rsubtil** [rsubtil/controller_icons#66]
https://github.com/godotengine/godot/pull/78539 added a way to get more "raw" information about a controller. In particular, it gives access to vendor/product ID on Linux.

Since there are public databases (such as https://devicehunt.com) documenting this information, controller model detection on Linux can be made much more reliable.

**Comment by mubinulhaque** [comment]
On the one hand, I like this idea because it could bypass things like Steam Input. On the other hand, it can cause issues when companies publish revisions of existing controllers. For examples, some games like Dying Light support PS4 buttom prompts but ONLY when the DS4 V1 controller is used. This controller is no longer manufactured and you have to use DS4Windows to spoof the software.

**Comment by rsubtil** [comment]
> On the one hand, I like this idea because it could bypass things like Steam Input. On the other hand, it can cause issues when companies publish revisions of existing controllers. For examples, some games like Dying Light support PS4 buttom prompts but ONLY when the DS4 V1 controller is used. This controller is no longer manufactured and you have to use DS4Windows to spoof the software.

Yeah, controller revisions make this much harder, thus why I'm keeping the idea now as backlog until I do more research/receive feedback. I can say though that I'd definitely only use this mechanism if it is more reliable, and keep the fallback to the controller name the OS reports.

**Comment by offalynne** [comment]
Best approach is a 2 stage heuristic:
1. Vendor product lookup
2. Fuzzy name match

SDL source provides a start: https://github.com/libsdl-org/SDL/blob/main/src/joystick/controller_list.h
I maintain an additive list based on the [Community Mapping DB](https://github.com/mdqinc/SDL_GameControllerDB) Godot uses (which I also comaintain)

Name match is a bit more involved, but similar sources are available.

I'm interested in breaking the above data out into a separate repository, if there's interest
